src/libmongoc/tests/test-mongoc-uri.c

```diff
@@ -705,52 +650,29 @@ test_mongoc_uri_functions (void)
 }
 
 static void
-test_mongoc_uri_new_with_error (void)
+run_new_with_error_tests (const char *uri_str,
+			  const char *err_msg)
 {
    bson_error_t error = {0};
+
+   ASSERT (!mongoc_uri_new_with_error (uri_str, &error));
+   ASSERT_ERROR_CONTAINS (error,
+			  MONGOC_ERROR_COMMAND,
+			  MONGOC_ERROR_COMMAND_INVALID_ARG,
+			  err_msg);
+}
+
+static void
+test_mongoc_uri_new_with_error (void)
+{
    mongoc_uri_t *uri;
 
-   capture_logs (true);
-   ASSERT (!mongoc_uri_new_with_error ("mongodb://", NULL));
+   /* Test a valid URI */
    uri = mongoc_uri_new_with_error ("mongodb://localhost", NULL);
    ASSERT (uri);
    mongoc_uri_destroy (uri);
 
-   ASSERT (!mongoc_uri_new_with_error ("mongodb://", &error));
-   ASSERT_ERROR_CONTAINS (error,
-                          MONGOC_ERROR_COMMAND,
-                          MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "Invalid host string in URI");
-
-   memset (&error, 0, sizeof (bson_error_t));
-   ASSERT (!mongoc_uri_new_with_error ("mongo://localhost", &error));
-   ASSERT_ERROR_CONTAINS (error,
-                          MONGOC_ERROR_COMMAND,
-                          MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "Invalid URI Schema, expecting 'mongodb://'");
-
-   memset (&error, 0, sizeof (bson_error_t));
-   ASSERT (!mongoc_uri_new_with_error (
-      "mongodb://localhost/?readPreference=unknown", &error));
-   ASSERT_ERROR_CONTAINS (
-      error,
-      MONGOC_ERROR_COMMAND,
-      MONGOC_ERROR_COMMAND_INVALID_ARG,
-      "Unsupported readPreference value [readPreference=unknown]");
-
-   memset (&error, 0, sizeof (bson_error_t));
-   ASSERT (!mongoc_uri_new_with_error (
-      "mongodb://localhost/"
-      "?appname="
-      "WayTooLongAppnameToBeValidSoThisShouldResultInAnErrorWayToLongAppnameToB"
-      "eValidSoThisShouldResultInAnErrorWayToLongAppnameToBeValidSoThisShouldRe"
-      "sultInAnError",
-      &error));
-   ASSERT_ERROR_CONTAINS (error,
-                          MONGOC_ERROR_COMMAND,
-                          MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "Unsupported value for \"appname\""); /* ... */
-
+   // TODO Sam move this test, it is not testing new_with_error.
```

Comment:
It looks like this was moved in L691. I think you can remove this and the immediate test below.